### PR TITLE
Qt: Add option to invoke the updater manually

### PR DESCRIPTION
**Pull request by spycrab**
Screenshot

### CODE REVIEW

File: Source/Core/DolphinQt2/MenuBar.cpp [hunk 1]
```diff
@@ -490,9 +491,29 @@ void MenuBar::AddOptionsMenu()
   m_change_font = AddAction(options_menu, tr("&Font..."), this, &MenuBar::ChangeDebugFont);
 }
 
+void MenuBar::InstallUpdateManually()
+{
+  auto& track = SConfig::GetInstance().m_auto_update_track;
+  auto previous_value = track;
+
+  track = "dev";
```
Comment (leoetlino) [inline Source/Core/DolphinQt2/MenuBar.cpp]: Do we really want to force an update to the latest dev version even for people who have selected another track?

Comment (spycrab) [reply Source/Core/DolphinQt2/MenuBar.cpp]: Well it's probably correct to assume they want to install the latest version if they choose to check for updates. Furthermore most people using this feature will not have a track selected anyway.

File: Source/Core/DolphinQt2/MenuBar.cpp [hunk 2]
```diff
@@ -490,9 +491,29 @@ void MenuBar::AddOptionsMenu()
...
+
+  auto* updater = new Updater(this);
+
+  if (!updater->CheckForUpdate())
+  {
+    QMessageBox::information(
+        this, tr("Update"),
+        tr("You are running the latest version available on this update track"));
```
Comment (riking) [inline Source/Core/DolphinQt2/MenuBar.cpp]: This message is wrong if you're switching track to dev before checking.

**Comment by Antidote** [comment]
This is outside the scope of this PR, however I feel that the updater should be opt-in at compile-time on Linux/BSD based distributions as, for the most, updates are managed by the distribution's package managers, or their community (Arch's AUR for example).

**Comment by spycrab** [comment]
@Antidote It's neither supported nor enabled on any platform but Windows.

**Pull request merged**